infrastructure/pipeline_storage/README.md

```diff
@@ -0,0 +1,33 @@
+# pipeline_storage
+
+This stack contains the pipeline_storage cluster we use for intermediate storage in the pipeline.
+
+Why does this live in its own stack?
+
+*   It's not in `infrastructure/critical` because it's an ephemeral resource.
+    It'd be annoying if we lost it, but we can recreate all the data it contains.
+
+    We keep it separate from more important resources -- say, the ID minter database -- to reduce the risk of a bad Terraform change hosing something we really care about.
+
+*   It's not in `pipeline` because it has to be managed using the `run_terraform.sh` script to get Elastic Cloud credentials.
+
+    This is moderately inconvenient -- you can't use it by running vanilla Terraform.
+    For now I don't want to make the pipeline stack harder to use.
+
+## Scaling up the cluster during reindexes
+
+We can't afford to run a high-capacity cluster all the time, but we also don't need to.
+The cluster is under heavy load when we're reindexing the catalogue pipeline, but that's it.
+
+*   When the pipeline is idle, we leave it at a single 8GB node.
+*   When we're reindexing, it's helpful to turn it up to 29GB and add an extra zone.
+
+Remember to turn it down again afterwards!
```

Comment:
What about a "reindexing" variable on the stack that takes care of sizeing for you?

Reply:
I think we could do that if we moved the pipeline_storage module into the pipeline stack.

How do we feel about that?

Reply:
So implicitly that means one ES cluster per pipeline, which would make the base cost n*minimum_cost_of_es_cluster. I think maybe. Let's discuss elsewhere.

Reply:
When I wrote that I actually meant defining the cluster at the top of the pipeline stack (still a single instance, but living alongside it), but that doesn't really work with targeted plan/apply.

It also requires some thinking about authentication. I think it's possible, but agree we should discuss further.